server/src/main/java/org/apache/druid/segment/realtime/firehose/HttpFirehoseFactory.java

```diff
@@ -54,7 +59,9 @@ public HttpFirehoseFactory(
       @JsonProperty("maxFetchCapacityBytes") Long maxFetchCapacityBytes,
       @JsonProperty("prefetchTriggerBytes") Long prefetchTriggerBytes,
       @JsonProperty("fetchTimeout") Long fetchTimeout,
-      @JsonProperty("maxFetchRetry") Integer maxFetchRetry
+      @JsonProperty("maxFetchRetry") Integer maxFetchRetry,
+      @JsonProperty("httpAuthenticationUsername") String httpAuthenticationUsername,
+      @JsonProperty("httpAuthenticationPassword") DefaultPasswordProvider httpAuthenticationPasswordProvider
```

Comment:
Any reason to allow only `DefaultPasswordProvider`? The intention of using `PasswordProvider` is to provide `DefaultPasswordProvider` by default, but users can use `EnvironmentVariablePasswordProvider` or their own custom passwordProvider if they want to hide password information from logs.

Reply:
Also please add `@Nullable`.

Reply:
Thank you for calling this out. Now that I understand PasswordProvider more I get why using the vanilla `PasswordProvider` is best. I can see the environment variable being useful for teams who don't want to be sending passwords in the JSON